### PR TITLE
newcomputermodern: init at 4.4

### PR DESCRIPTION
**Pull request by drupol**
`newcomputermodern` font from https://ctan.org/pkg/newcomputermodern?lang=en

Related to https://github.com/typst/typst/pull/340

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/data/fonts/newcomputermodern/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{ lib, stdenvNoCC, fetchzip }:
+
+stdenvNoCC.mkDerivation rec {
+  pname = "newcomputermodern";
+  version = "4.4";
+
+  src = fetchzip {
+    url = "https://mirrors.ctan.org/fonts/newcomputermodern.zip";
```
Comment (apfelkuchen6) [inline pkgs/data/fonts/newcomputermodern/default.nix:8]: This is not a stable url. Maybe use https://ftp.math.utah.edu/pub/tex/historic/systems/texlive/2022/tlnet-final/archive/newcomputermodern.doc.r66338.tar.xz (or another mirror from https://tug.org/historic/ ) instead?

newcomputermodern is also already package as part of texlive, so just repackaging this in a more convenient format (similar to https://github.com/NixOS/nixpkgs/blob/master/pkgs/data/fonts/iwona/default.nix ) is also an option.

Comment (drupol) [reply pkgs/data/fonts/newcomputermodern/default.nix]: Excellent ! I didn't know that. I updated the PR.

File: pkgs/data/fonts/newcomputermodern/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
+{ lib, stdenvNoCC, fetchzip, texlive }:
+
+stdenvNoCC.mkDerivation rec {
+  pname = "newcomputermodern";
+  version = "4.4";
```
Comment (apfelkuchen6) [inline pkgs/data/fonts/newcomputermodern/default.nix:5]:
```suggestion
  inherit (src) version;
```
Avoiding drifts of the version number and the actual version is probably also a good idea. (This also applies to iwona).

Comment (drupol) [reply pkgs/data/fonts/newcomputermodern/default.nix]: Fixed!

**Comment by kanashimia** [comment]
Note that New Computer Modern is already in the nixpkgs, you can try `fonts.fonts = pkgs.texlive.newcomputermodern.pkgs;` on NixOS, it is a list of derivations, but works.
I think it is automatically generated from the texlive deps.

**Comment by drupol** [comment]
Let me try immediately

**Comment by drupol** [comment]
Closing the PR, the font is available through `pkgs.texlive.newcomputermodern.pkgs` indeed.

Thanks!!!!

**Comment by kanashimia** [comment]
@drupol it would be fine to just repackage this from texlive, as per @apfelkuchen6 suggestion, you was too eager to delete the branch :)
This is slightly more convenient that it being a list, and would adhere to the output directory structure.
After all some other fonts are packaged two times too, like libertine that typst uses.

**Comment by LU15W1R7H** [comment]
I would also like for this to be a separate package.

**Comment by drupol** [comment]
> I would also like for this to be a separate package, especially for use with typst.

Typst has `newcomputermodern` embedded by default in it.

> @drupol it would be fine to just repackage this from texlive, as per @apfelkuchen6 suggestion, you was too eager to delete the branch :)
> This is slightly more convenient that it being a list, and would adhere to the output directory structure.
> After all some other fonts are packaged two times too, like libertine that typst uses.

I just tried to use the following without success:

```nix
  fonts.packages = [
    pkgs.texlive.newcomputermodern.pkgs
  ];
```

Error:

<details>

```
       … while calling 'check'

         at /nix/store/55ql4j8d47xjvfa8xggjddgfwny4n9j7-source/lib/types.nix:510:15:

          509|       descriptionClass = "noun";
          510|       check = x: isStringLike x && builtins.substring 0 1 (toString x) == "/";
             |               ^
          511|       merge = mergeEqualOption;

       … while evaluating derivation 'fontconfig-conf'
         whose name attribute is located at /nix/store/55ql4j8d47xjvfa8xggjddgfwny4n9j7-source/pkgs/stdenv/generic/make-derivation.nix:348:7

       … while evaluating attribute 'buildCommand' of derivation 'fontconfig-conf'

         at /nix/store/55ql4j8d47xjvfa8xggjddgfwny4n9j7-source/pkgs/build-support/trivial-builders/default.nix:87:14:

           86|       enableParallelBuilding = true;
           87|       inherit buildCommand name;
             |              ^
           88|       passAsFile = [ "buildCommand" ]

       … while evaluating derivation 'fc-00-nixos-cache.conf'
         whose name attribute is located at /nix/store/55ql4j8d47xjvfa8xggjddgfwny4n9j7-source/pkgs/stdenv/generic/make-derivation.nix:348:7

       … while evaluating attribute 'text' of derivation 'fc-00-nixos-cache.conf'

         at /nix/store/55ql4j8d47xjvfa8xggjddgfwny4n9j7-source/pkgs/build-support/trivial-builders/default.nix:148:16:

          147|     runCommand name
          148|       { inherit text executable checkPhase allowSubstitutes preferLocalBuild;
             |                ^
          149|         passAsFile = [ "text" ];

       … while calling anonymous lambda

         at /nix/store/55ql4j8d47xjvfa8xggjddgfwny4n9j7-source/nixos/modules/config/fonts/fontconfig.nix:54:39:

           53|         <!-- Font directories -->
           54|         ${concatStringsSep "\n" (map (font: "<dir>${font}</dir>") config.fonts.packages)}
             |                                       ^
           55|         ${optionalString (pkgs.stdenv.hostPlatform == pkgs.stdenv.buildPlatform) ''

       … while calling anonymous lambda

         at /nix/store/55ql4j8d47xjvfa8xggjddgfwny4n9j7-source/lib/types.nix:528:14:

          527|       merge = loc: defs:
          528|         map (x: x.value) (filter (x: x ? value) (concatLists (imap1 (n: def:
             |              ^
          529|           imap1 (m: def':

       error: A definition for option `fonts.packages."[definition 1-entry 6]"' is not of type `path'. Definition values:
       - In `/nix/store/2rk77kqp7x6y33gcy9mxwhrvnklggf6b-source/hosts/common/fonts.nix':
           [
             <derivation texlive-newcomputermodern-4.4>
             <derivation texlive-newcomputermodern.doc-4.4>
           ]
```

<details>

Do you think we should repackage this as a regular derivation?

**Comment by drupol** [comment]
Follow up in #280060

**Pull request closed without merging**